### PR TITLE
Merge authentication functions into a single middleware

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
- Merged passport config/initialisation from auth.ts into `setupAuthenticationMiddleware`
- Merged `authenticationMiddleware(...)` middleware from auth.ts into `setupAuthenticationMiddleware`
- Removed auth.ts

### CODE REVIEW

File: server/middleware/setUpAuthentication.ts [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
+    (token, refreshToken, params, profile, done) => {
+      return done(null, { token, username: params.user_name, authSource: params.auth_source })
+    },
+  ),
+)
+
+export default function setupAuthentication() {
+  const router = Router()
+  const authUrl = config.apis.hmppsAuth.externalUrl
+  const authParameters = `client_id=${config.apis.hmppsAuth.authClientId}&redirect_uri=${config.domain}`
```
Comment (andrewrlee) [inline server/middleware/setUpAuthentication.ts:40]: did you move these for any reason? [it's generally good practice to declare variables close to where they're used](https://stackoverflow.com/a/33246061).

Comment (Thomas-Geraghty) [reply server/middleware/setUpAuthentication.ts:40]: I think I'm used to the second example given in there, that's just a personal preference of mine 😄 I'll shift these back

Comment (Thomas-Geraghty) [reply server/middleware/setUpAuthentication.ts]: Resolved in https://github.com/ministryofjustice/hmpps-template-typescript/pull/413/commits/0755dfd828172be677a0ac16272efca07c95ce71

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**